src/include/cpu/power/scom.h

```diff
@@ -85,26 +86,53 @@ typedef enum
 	EC23_CHIPLET_ID = 0x37      ///< Core23 chiplet (Quad5, EX11, C1)
 } chiplet_id_t;
 
+void reset_scom_engine(void);
+
 static uint64_t read_scom_direct(uint64_t reg_address)
 {
 	uint64_t val;
-	asm volatile(
-		"ldcix %0, %1, %2":
-		"=r"(val):
-		"b"(0x800603FC00000000),
-		"r"(reg_address << 3));
-	eieio();
+	uint64_t hmer;
+	do {
+		clear_hmer();
+		eieio();
+		asm volatile(
+			"ldcix %0, %1, %2":
+			"=r"(val):
+			"b"(0x800603FC00000000),
```

Comment:
This address 0x800603FC00000000 could be defined as SCOM base or something

Reply:
Added, although this will have to be changed after we add support for second CPU.